op-batcher/batcher/driver.go

```diff
@@ -835,22 +835,34 @@ func (l *BatchSubmitter) sendTransaction(txdata txData, queue *txmgr.Queue[txRef
 	if *candidate.To != l.RollupConfig.BatchInboxAddress {
 		return fmt.Errorf("candidate.To is not inbox")
 	}
+
+	l.sendTx(txdata, false, candidate, queue, receiptsCh)
+	return nil
+}
+
+// sendTx uses the txmgr queue to send the given transaction candidate after setting its
+// gaslimit. It will block if the txmgr queue has reached its MaxPendingTransactions limit.
+func (l *BatchSubmitter) sendTx(txdata txData, isCancel bool, candidate *txmgr.TxCandidate, queue *txmgr.Queue[txRef], receiptsCh chan txmgr.TxReceipt[txRef]) {
 	isEOAPointer := l.inboxIsEOA.Load()
 	if isEOAPointer == nil {
 		ctx, cancel := context.WithTimeout(context.Background(), time.Second*2)
 		defer cancel()
 		var code []byte
-		code, err = l.L1Client.CodeAt(ctx, *candidate.To, nil)
+		code, err := l.L1Client.CodeAt(ctx, *candidate.To, nil)
 		if err != nil {
-			return fmt.Errorf("CodeAt failed:%w", err)
+			l.Log.Error("CodeAt failed, assuming code exists", "err", err)
+			// assume code exist, but don't persist the result
+			isEOA := false
+			isEOAPointer = &isEOA
+		} else {
+			isEOA := len(code) == 0
+			isEOAPointer = &isEOA
+			l.inboxIsEOA.Store(isEOAPointer)
 		}
-		isEOA := len(code) == 0
-		isEOAPointer = &isEOA
-		l.inboxIsEOA.Store(isEOAPointer)
-	}
 
+	}
 	// Don't set GasLimit when inbox is contract so that later on `EstimateGas` will be called
```

Comment:
I feel the comments here should be updated to 
```suggestion
	// Set GasLimit as intrinstic gas if the inbox is EOA, otherwise
	// Leave GasLimit unset when inbox is contract so that later on `EstimateGas` will be called
```

Reply:
Done in [7aab5ae](https://github.com/ethstorage/optimism/pull/96/commits/7aab5ae1fe87fd4f69442fcb7bbdb9076bc8b954)

Reply:
Done in [7aab5ae](https://github.com/ethstorage/optimism/pull/96/commits/7aab5ae1fe87fd4f69442fcb7bbdb9076bc8b954).